mico-core/src/main/java/io/github/ust/mico/core/REST/ApplicationController.java

```diff
@@ -121,18 +121,22 @@
                 linkTo(methodOn(ApplicationController.class).getApplicationsByShortName(shortName)).withSelfRel()));
     }
 
-    @PutMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}/services")
-    public ResponseEntity addServiceToApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String _shortName,
-                                                                         @PathVariable(PATH_VARIABLE_VERSION) String _version,
-                                                                         @RequestBody MicoService _service) {
-        Optional<MicoService> serviceOptional = serviceRepository.findByShortNameAndVersion(_service.getShortName(), _service.getVersion());
-        Optional<MicoApplication> applicationOptional = applicationRepository.findByShortNameAndVersion(_shortName, _version);
+    @PostMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}/services")
+    public ResponseEntity addServiceToApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String applicationShortName,
+                                                  @PathVariable(PATH_VARIABLE_VERSION) String applicationVersion,
+                                                  @RequestBody MicoService serviceFromBody) {
+        Optional<MicoService> serviceOptional = serviceRepository.findByShortNameAndVersion(serviceFromBody.getShortName(), serviceFromBody.getVersion());
+        Optional<MicoApplication> applicationOptional = applicationRepository.findByShortNameAndVersion(applicationShortName, applicationVersion);
         if (serviceOptional.isPresent() && applicationOptional.isPresent()) {
-            MicoService newService = serviceOptional.get();
+            MicoService service = serviceOptional.get();
             MicoApplication application = applicationOptional.get();
-            MicoApplication applicationWithService = application.toBuilder().service(newService).build();
-            applicationRepository.save(applicationWithService);
-            return ResponseEntity.noContent().build();
+            if (!application.getServices().contains(service)) {
+                MicoApplication applicationWithService = application.toBuilder().service(service).build();
+                applicationRepository.save(applicationWithService);
+                return ResponseEntity.noContent().build();
+            } else {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The application already contains this service");
```

Comment:
Why not make adding the same service a no-op? This would make the call idempotent.

Reply:
I agree but providing meeningful information for the user could be left to the ui in this case. A user that pressed a button twice would not want to see an error. Which is easy to implement if the function is idempotent but hard if not. And the ui will filter existing services out from the possible choices.

Reply:
I suggest to not go too much into detail - remember the sprint goal (minimal example). I would suggest to just add a `TODO` and then check it later again (and fix it).